scan/dependencies.go

```diff
@@ -34,7 +34,9 @@ func (s *Scanner) ScanForDependencies(context string, workingDir string, dockerf
 	// In the case of remote contexts, it's scoped to the clone or downloaded location and the working directory.
 	// For example, for a git URL ending with .git#:foo/bar which was downloaded to a directory "build",
 	// the path is scoped to build/foo/bar/Dockerfile.
-	if !util.IsLocalContext(s.context) {
+	// If the Dockerfile was not specified, we default its value to "Dockerfile" and check it relative
+	// to the context.
+	if s.defaultedDockerfile || !util.IsLocalContext(s.context) {
```

Comment:
>defaultedDockerfile [](start = 6, length = 19)

If you just pass empty dockerfile to scanner, you can check here and assign dockerfilePath to `path.Clean(path.Join(workingDir, "Dockerfile"))`

Therefore you can avoid introducing the new `defaultedDockerfile` parameter.